apps/api/src/app/bridge/usecases/store-control-variables/store-control-variables.usecase.ts

```diff
@@ -40,60 +26,24 @@ export class StoreControlVariables {
 
     const step = workflowExist?.steps.find((item) => item.stepId === command.stepId);
 
-    if (!step) {
+    if (!step || !step._id) {
       throw new ApiException('Step not found');
     }
 
-    const stepDefault = defaults(
+    const stepDefaultControls = defaults(
       (step.template as any)?.controls?.schema || (step.template as any)?.inputs?.schema,
       {}
+    ) as JsonSchema;
```

Comment:
GAL: Concerned the type is not real and can cause runtime issues, please revise @djabarovgeorge

Reply:
@tatarco can you elaborate on the runtime issues experienced with this?

Reply:
there is an assumption here and a casting done without any checking, this was a sync PR comment for george to make sure it's always true because if not it's an assumption that can break lots of stuff
   const stepDefaultControls = defaults(
      (step.template as any)?.controls?.schema || (step.template as any)?.inputs?.schema,
      {}
    ) as JsonSchema;